### PR TITLE
Gradient checkpointing

### PR DESCRIPTION
**Pull request by joehoover**
This pull request introduces the following changes: 

1. **Gradient checkpointing is turned on by default.** It can be controlled with the bool `gradient_checkpointing`. Using gradient checkpointing dramatically reduces GPU memory usage. In fact, the effect is so substantial that it allows us to turn off CPU offloading and comfortably increase batch sizes by about a factor of 4. 

2. **Turn off CPU offloading.** By default, a new DeepSpeed config is used for training. Still ZeRO-3, but without CPU offloading.

3. **No gradient accumulation by default.** Because we can use larger batch sizes, gradient accumulation is turned off by default (e.g. default value is now `1` instead of `8`). 

4. Default logging steps changed from `1` to `100`.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -53,13 +53,18 @@ def train(
         default=-1
     ),
     logging_steps: int = Input(
-        description="number of steps between logging epoch & loss", default=1
+        description="number of steps between logging epoch & loss", default=100
```
Comment (daanelson) [inline train.py:56]: curious what the latency impact would be if we lowered this - in general I think it's a great idea to have a high # of steps here, but since we have no other way of surfacing loss curves to users atm, perhaps we should keep this lower?

Comment (joehoover) [reply train.py]: Lowered!
